core/matrix/conv.cpp

```diff
@@ -50,7 +51,22 @@ template <typename ValueType>
 void Conv<ValueType>::validate_application_parameters(const LinOp* b,
                                                       const LinOp* x) const
 {
-    // implement dimension validation throw DimensionMismatch when it is wrong
+    using gko::detail::get_size;
+    const auto b_rows = get_size(b)[0];
+    const auto x_rows = get_size(x)[0];
+    const auto kernel_len = kernel_.get_size();
+
+    if (x_rows != b_rows + kernel_len - 1) {
+        throw DimensionMismatch(
+            __FILE__, __LINE__, __func__, "x", x_rows, 1,
+            "(b + 2*padding - kernel)/stride + 1",
+            (b_rows + 2 * 2 - kernel_len) / 1 + 1, 1,
```

Comment:
```suggestion
            (b_rows + 2 * 0 - kernel_len) / 1 + 1, 1,
```
shouldn't padding be zero now?

Reply:
padding equal 2 in your example such that it become numpy result.
IMO, we should follow the default from torch not numpy.
also, if it needs to fit numpy, it should be something like ((x-1)*stride + kernel - b)/2